demisto_sdk/commands/common/hook_validations/base_validator.py

```diff
@@ -130,3 +132,71 @@ def _is_id_equals_name(self, file_type):
+        tests = self.current_file.get('tests', [])
+        if not self.yml_has_test_key(tests, file_type):
+            return False
+        return self.tests_registered_in_conf_json_file(tests)
+
+    def tests_registered_in_conf_json_file(self, tests):
+        # If 'No tests' is explicitly configured, we can skip checking if the tests are
+        # registered in the conf.json file.
+        no_tests_explicitly = any(test for test in tests if 'no test' in test.lower())
+        if no_tests_explicitly:
+            return True
+        conf_json__tests = self._load_conf_file()['tests']
+
+        # Calling it integration but the validation is the same for integrations, playbooks and scripts
+        integration_id = self._get_file_id(self.structure_validator.scheme_name)
+        not_registered_tests = self.get_not_registered_tests(conf_json__tests, integration_id, tests)
+        if not_registered_tests:
+            missing_test_configurations = json.dumps([
+                {'integrations': integration_id, 'playbookID': test} for test in not_registered_tests
+            ], indent=4).strip('[]')
+            error_message = \
+                f'The following TestPlaybooks are not registered in {self.CONF_PATH} file.\n' \
+                f'Please add\n{missing_test_configurations}\nto {self.CONF_PATH} path under \'tests\' key ' \
+                f'or run \'demisto-sdk format --skip-test-playbooks -p {self.file_path}\''
+            print_error(error_message)
+            return False
+        return True
+
+    def get_not_registered_tests(self, conf_json_data, integration_id, tests):
+        not_registered_tests = []
+        for test in tests:
+            test_registered_in_conf_json = any(
+                test_config for test_config in conf_json_data if self.find_test_match(test_config, test, integration_id)
+            )
+            if not test_registered_in_conf_json:
+                not_registered_tests.append(test)
+        return not_registered_tests
+
+    def yml_has_test_key(self, tests, file_type):
+        if not tests:
+            print_error(
+                f'You don\'t have a TestPlaybook for {file_type} {self.file_path}. If you don\'t want to create a'
+                f' TestPlaybook for this {file_type}, edit the yml file and add  \ntests:\n -  No tests\n lines'
+                f' to it. Or run demisto-sdk format --skip-test-playbooks -p {file_type}')
+            return False
+        return True
+
+    @staticmethod
+    def find_test_match(test_config, test_playbook_id, integration_id):
+        test_integrations = test_config.get('integrations')
+        if test_playbook_id != test_config.get('playbookID'):
+            return False
+        if isinstance(test_integrations, list):
+            return any(
+                test_integration for test_integration in test_integrations if test_integration == integration_id)
+        else:
+            return test_integrations == integration_id
```

Comment:
I believe this logic is incorrect. As we said, test playbooks aren't only linked to integrations. With that understanding, in the `conf.json` we have test "configurations" that do not contain an `"integrations"` key... because they don't test an integration, they might be tests for a script or another playbook.